File: backend/src/main/java/io/ntf/api/statistics/model/UserConfiguration.java
package io.ntf.api.statistics.model;

import lombok.*;
import lombok.experimental.Wither;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "userConfigurations")
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Wither
public class UserConfiguration {

  @Id
  private String id;
  private String userId;
  private WorkTimeConfiguration workTimeConfiguration;

}

File: backend/src/main/java/io/ntf/api/statistics/model/UserConfiguration.kt
package io.ntf.api.statistics.model

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document(collection = "userConfigurations")
data class UserConfiguration(@Id val id: String? = null,
                             val userId: String,
                             val workTimeConfiguration: WorkTimeConfiguration)
